src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineComputer.java

```diff
@@ -117,9 +130,18 @@ public HttpResponse doDoDelete() throws IOException {
         ComputeEngineInstance node = getNode();
         if (node != null) {
             try {
+                ComputeEngineCloud cloud = getCloud();
+
+                // Checks for failed jobs for this computer's node
+                if (!this.getBuilds().failureOnly().isEmpty()) {
+                    createSnapshot(cloud);
+                }
+
                 node.terminate();
             } catch (InterruptedException ie) {
-                //TODO: log
+                LOGGER.info("Interrupted exception: " + ie);
```

Comment:
No need to spell out the exception type in the log message, that's included in the toString(). Rather focus on providing the user with details as to what caused the exception, like mentioning the snapshotting.

Reply:
And I think would be nice to pass full stacktrace to logger too.. (second parameter exception)

Reply:
Not sure what you mean by full stacktrace when I already give logger the exception.

Reply:
The Logger class provides an overload of log which allows you to pass the throwable as an argument: https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#log(java.util.logging.Level,%20java.lang.String,%20java.lang.Throwable)